src/elements/content-picker/stories/tests/ContentPicker-visual.stories.js

```diff
@@ -17,8 +25,204 @@ export const selectedEmptyState = {
+                    return HttpResponse.error();
+                }),
+            ],
+        },
+    },
+};
+
+export const hitSelectionLimit = {
+    args: {
+        maxSelectable: 2,
+    },
+    parameters: {
+        msw: {
+            handlers: [
+                http.get(`${DEFAULT_HOSTNAME_API}/2.0/folders/69083462919`, () => {
+                    return HttpResponse.json(mockRootFolder);
+                }),
+            ],
+        },
+    },
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+
+        // Wait for items to load and select first item
+        await waitFor(() => {
+            const items = canvas.getAllByRole('row');
+            expect(items.length).toBeGreaterThan(1); // Header row + at least one item
+        });
+
+        const items = canvas.getAllByRole('row');
+        await userEvent.click(items[1]); // Select first item after header
+
+        await waitFor(() => {
+            expect(canvas.getByText('1 Selected')).toBeInTheDocument();
+            const chooseButton = canvas.getByLabelText('Choose');
+            expect(chooseButton).toBeEnabled();
+        });
+
+        // Select second item to hit limit
+        await userEvent.click(items[2]); // Select second item
+        await waitFor(() => {
+            expect(canvas.getByText('2 Selected')).toBeInTheDocument();
+            expect(canvas.getByText('(max)')).toBeInTheDocument();
+            const chooseButton = canvas.getByLabelText('Choose');
+            expect(chooseButton).toBeEnabled();
+        });
+
+        // Try to select beyond limit
+        if (items.length > 3) {
+            await userEvent.click(items[3]); // Try to select third item
+            await waitFor(() => {
+                expect(canvas.getByText('2 Selected')).toBeInTheDocument();
+                expect(canvas.getByText('(max)')).toBeInTheDocument();
+                const chooseButton = canvas.getByLabelText('Choose');
+                expect(chooseButton).toBeEnabled();
+            });
+        }
+    },
+};
+
+export const singleSelectWithItems = {
+    args: {
+        maxSelectable: 1,
+    },
+    parameters: {
+        msw: {
+            handlers: [
+                http.get(`${DEFAULT_HOSTNAME_API}/2.0/folders/69083462919`, () => {
+                    return HttpResponse.json(mockRootFolder);
+                }),
+            ],
+        },
+    },
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+
+        // Wait for items to load
+        await waitFor(() => {
+            const items = canvas.getAllByRole('row');
+            expect(items.length).toBeGreaterThan(1); // Header row + at least one item
         });
+
+        const items = canvas.getAllByRole('row');
+
+        // Select first item
+        await userEvent.click(items[1]); // Select first item after header
+        await waitFor(() => {
+            expect(canvas.getByText('1 Selected')).toBeInTheDocument();
```

Comment:
when max is 1, the selected ui does not appear, it changed to radio buttons where you cant select more than one radio button

Reply:
what about this?